application-gardening-ui/src/main/resources/Gardening/Code/GardeningTranslations.fr.xml

```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.3" reference="Gardening.Code.GardeningTranslations" locale="fr">
+  <web>Gardening.Code</web>
+  <name>GardeningTranslations</name>
+  <language>fr</language>
+  <defaultLanguage>en</defaultLanguage>
+  <translation>1</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <creationDate>1525552355000</creationDate>
+  <parent>Gardening.Code.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <date>1525554234000</date>
+  <contentUpdateDate>1525554183000</contentUpdateDate>
+  <version>1.1</version>
+  <title>GardeningTranslations</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>plain/1.0</syntaxId>
+  <hidden>true</hidden>
+  <content>gardening.wiki.selectScripts=Sélectionnez les scripts de jardinage actifs
+gardening.wiki.startJob=Démarrer le jardinage
+gardening.wiki.start=Jardiner !
+gardening.job.success=Terminé !
+gardening.job.error=Une erreur est survenue, consultez les journaux pour plus d'information.
+Gardening.Code.GardeningScriptConfigurationClass_activeQueryScripts=Scripts de requête actifs
+Gardening.Code.GardeningScriptConfigurationClass_activeActionScript=Script d'action actif</content>
```

Comment:
is there a problem of lowercase/uppercase. The app name is not consistent. It should be the lowercase version which is good.

Reply:
See also http://dev.xwiki.org/xwiki/bin/view/Community/L10N%20Conventions

Reply:
IMO it's more a problem of the L10N practices that are not consistent with the reality of contrib applications. 

Here are two examples : 
* https://github.com/xwiki-contrib/application-blog/tree/master/application-blog-ui/src/main/resources/Blog
* https://github.com/xwiki-contrib/application-forum/tree/master/application-forum-ui/src/main/resources

Reply:
So here, `Gardening` keys refers to page names and are used to translate class properties. 
Personally, I've never seen a wiki page that belongs to an application and that starts with a non-capital letter.

Moreover, the two examples I gave above (https://github.com/xwiki-contrib/application-gardening/pull/1#discussion_r188912541) show applications that uses capital letters in their translation keys (see https://github.com/xwiki-contrib/application-forum/blob/f969f31003522febd88bc6a2a0b45f232c2cbd34/application-forum-ui/src/main/resources/ForumCode/Translations.xml#L43).

Reply:
Can you tell me where we have a rule that says that a translation needs to use the wiki page name for the prefix? (I don't recall this).

Your example 1 link for the forum is bad since that clearly doesn't follow the best practices.

Reply:
I vaguely remember some naming rules that we need to follow for ConfigurableClass but I don't recall the details.

Maybe you can explain the context where you use your translations with uppercases?

Reply:
Found one here: https://github.com/xwiki-contrib/application-gardening/blob/f2135107f13e8691ba495d4ba056311943859311/application-gardening-ui/src/main/resources/Gardening/Code/GardeningScriptAdministration.xml#L61 

This is clearly not following any best practice I know :) But maybe I'm missing something so any link would help.

Reply:
In addition it's really a bad practice IMO since it ties a translation, to a wiki page location which will break as soon as you refactor the page and move code to another location...

Reply:
So, just to be sure about the problems we're having now, you don't think that translation keys used for translating class properties should have prefixes with capital letters ?

Reply:
We have a ton of examples like that in platform, here are two other ones : 
* https://github.com/xwiki/xwiki-platform/blob/fa832612e6025bca86a3c00e1e2667fe20747131/xwiki-platform-core/xwiki-platform-invitation/xwiki-platform-invitation-ui/src/main/resources/Invitation/Translations.xml#L45
* https://github.com/xwiki/xwiki-platform/blob/fa832612e6025bca86a3c00e1e2667fe20747131/xwiki-platform-core/xwiki-platform-annotations/xwiki-platform-annotation-ui/src/main/resources/AnnotationCode/Translations.xml#L110

Reply:
@vmassol this does not have much to do with naming best practice, that's just how class property translations works. When you want to provide a translation for a class property you have to follow a specific syntax `<class reference>_<property name> and the reference of the class here is `Gardening.Code.GardeningScriptConfigurationClass`. If you don't you won't see it in the object editor. See http://dev.xwiki.org/xwiki/bin/view/Community/L10N%20Conventions#HTranslatingXClasses.

Reply:
@tmortagne Yes you're right for class property translations but in the case of @aubincleme this is not the case. See the link I posted above: https://github.com/xwiki-contrib/application-gardening/blob/f2135107f13e8691ba495d4ba056311943859311/application-gardening-ui/src/main/resources/Gardening/Code/GardeningScriptAdministration.xml#L61

Reply:
To me it just mean this code reuse the standard activeQueryScripts class property translation. I don't see the point in providing two.

Reply:
ok I see what you mean. I didn't realize this translation key was used for different places.

Reply:
Ok so now the next step to discuss (not related to the gardening app) and long term is whether we thin it's a good idea to have translation keys named after a page (especially since that makes it hard to refactor the page). BTW just rearealizing that we may have broken lots of translations (and still are) when we moved code pages under the Code space...

FTR, the LT solves this by offering a translation prefix that can be specified by the dev.

Reply:
Posted to https://markmail.org/message/wu7qkuu7mevel5qp

Reply:
Ok, as this problem is resolved for now and for this issue ; merging …